mws/mws.py

```diff
@@ -76,18 +85,31 @@ def calc_request_description(params):
     """
     description_items = []
     for item in sorted(params.keys()):
-        encoded_val = quote(str(params[item]), safe='-_.~')
+        encoded_val = params[item]
         description_items.append('{}={}'.format(item, encoded_val))
     return '&'.join(description_items)
 
 
 def remove_empty(dict_):
-    """
-    Returns dict_ with all empty values removed.
-    """
     return {k: v for k, v in dict_.items() if v}
 
 
+def clean_extra_data(extra_data):
+    extra_data_enc = dict()
+    for key, value in extra_data.items():
+        if isinstance(value, (datetime.datetime, datetime.date)):
+            value = str(value.isoformat())
+        if isinstance(value, (bool, int)):
+            value = str(value)
+        try:
+            value.lower() + value + ''
+        except:
```

Comment:
flake8 fails the build due to a bare `except`, and then the test suite won't run at all. Aside from that, the test case for this exception is running `.lower()` and concatenating with `+`? Seems like an odd implementation.

Per another comment (regarding `isinstance(value, (bool, int))`), I think the proper `try` test is simply `value = str(value)`, but it's highly unlikely that that will even throw an exception of any kind. Thus, I don't know if a `try...except` here is even necessary.